File: gamelover-game-data-service/src/main/java/com/gamelover/gamedata/entity/Announcement.java
package com.gamelover.gamedata.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.Instant;

@Data
@Document(collection = "announcement")
public class Announcement {

    @Id
    private String id;

    @Field("announcement_id")
    private String announcementId;

    @Field("game_code")
    private String gameCode;

    @Field("title")
    private String title;

    @Field("content_html")
    private String contentHtml;

    @Field("tab_banner")
    private String tabBanner;

    @Field("fold_banner")
    private String foldBanner;

    @Field("is_red_dot")
    private Integer isRedDot;

    @Field("start_time_ms")
    private Long startTimeMs;

    @Field("end_time_ms")
    private Long endTimeMs;

    @Field("announcement_type")
    private String announcementType;

    @Field("priority")
    private Integer priority;

    @Field("create_time")
    private Instant createTime;

    @Field("update_time")
    private Instant updateTime;
}

File: frontend/src/stores/announcement.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'

export interface AnnouncementVO {
  announcementId: string
  gameCode: string
  title: string
  contentHtml: string
  tabBanner: string
  foldBanner: string
  isRedDot: number
  startTimeMs: number
  endTimeMs: number
  startTimeStr: string
  endTimeStr: string
  announcementType: string
  priority: number
  createTime: string
}

export const useAnnouncementStore = defineStore('announcement', () => {
  const announcements = ref<AnnouncementVO[]>([])
  const selectedAnnouncement = ref<AnnouncementVO | null>(null)
  const loading = ref(false)

  function setAnnouncements(list: AnnouncementVO[]) {
    announcements.value = list
  }

  function setSelectedAnnouncement(announcement: AnnouncementVO | null) {
    selectedAnnouncement.value = announcement
  }

  function setLoading(isLoading: boolean) {
    loading.value = isLoading
  }

  return {
    announcements,
    selectedAnnouncement,
    loading,
    setAnnouncements,
    setSelectedAnnouncement,
    setLoading
  }
})